std/algorithm/iteration.d

```diff
@@ -7950,6 +7956,14 @@ if (isRandomAccessRange!Range && hasLength!Range)
 
         next(2);
     }
+    static if (isForwardRange!Range)
```

Comment:
This is not necessary since `Range` is guaranteed at this point to be a [ForwardRange](https://github.com/dlang/phobos/pull/8580/files#diff-b13c31c11ad473d83762bcb8686d251f301236020f8751812580edcec49ef4c5R7895).